src/deployment/registration.py

```diff
@@ -85,18 +86,19 @@ def query_microsoft_graph(
 
 
 def retry(
-    operation: Callable[[], OperationResult],
+    operation: Callable[[Any], OperationResult],
```

Comment:
Since the type of `data` must match the first arg of `operation`, maybe this should use a generic type var that expresses this, rather than `Any`.

Reply:
Chose any for the simplicity. I Tried using a generic type but i was forced to convert the type to an optional and this was being propagated everywhere it was being called. This in turns forces the body of the method to check if the optional has a value to make mypy happy.

Reply:
Ah, I see, `mypy`'s type inference is doing quite what I'd want. I figured out how to work around it, but I think it is more confusing, on balance.